freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Controller.java

```diff
@@ -165,18 +166,18 @@ private String checkActorCrn() {
     private <T> T retryableWithDeletionException(CheckedFunction<Void, T, Exception> f) {
         try {
             return retryableFreeIpaClientService.invokeWithRetries(f);
-        } catch (DeleteException | InvalidFreeIpaStateException e) {
+        } catch (DeleteException | InvalidFreeIpaStateException | NotFoundException e) {
             throw e;
         } catch (RetryableFreeIpaClientException e) {
             LOGGER.error("internal server error", e);
             Exception exceptionForRestApi = e.getExceptionForRestApi();
             if (exceptionForRestApi instanceof DeleteException) {
                 throw (DeleteException) exceptionForRestApi;
             }
-            throw new DeleteException(DELETION_FAILED);
+            throw new DeleteException(DELETION_FAILED + e.getMessage());
```

Comment:
What is the difference between the `e` and the `exceptionForRestApi`. Maybe it better to throw with that message and log here the retry exception

Reply:
I had to check the code. We wrap most of the exceptions from the original one into a new one in case of KerberosMgmt. This means if we can't create keytab in FreeIPA, we would throw a `KeytabCreationException` instead of `FreeIpaClientException`. And we put this `KeytabCreationException` into `RetryableFreeIpaClientException` so our code would try it again, but if it fails multiple times we take the inner/original exception from it and return that to the API.

Reply:
So, would be better to add the original message to the exception, instead of the wrapped